bin/s2nc.c

```diff
@@ -535,16 +566,9 @@ int main(int argc, char *const *argv)
 
         /* Save session state from connection if reconnect is enabled */
         if (reconnect > 0) {
-            if (!session_ticket && s2n_connection_get_session_id_length(conn) <= 0) {
-                printf("Endpoint sent empty session id so cannot resume session\n");
-                exit(1);
-            }
-            free(session_state);
-            session_state_length = s2n_connection_get_session_length(conn);
-            session_state = calloc(session_state_length, sizeof(uint8_t));
-            if (s2n_connection_get_session(conn, session_state, session_state_length) != session_state_length) {
-                print_s2n_error("Error getting serialized session state");
-                exit(1);
+            /* Only need to wait to receive the session ticket in TLS1.3 */
+            if ((conn->actual_protocol_version >= S2N_TLS13 ) && session_ticket) {
+                echo(conn, sockfd, &session_ticket_recv);
```

Comment:
If echo returns an error, should we capture and exit or is echo always passing when we reach here?  I prefer the GUARD instead of exit(1) within echo that we do now.